src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -135,6 +138,10 @@ DataReaderImpl::DataReaderImpl(
     }
 
     att.times = qos.reliable_reader_qos().times;
+    att.liveliness_lease_duration = qos.liveliness().lease_duration;
+    att.liveliness_kind_ = qos.liveliness().kind;
+    att.matched_writers_allocation = qos.reader_resource_limits().matched_publisher_allocation;
```

Comment:
Should add disable_positive_acks here

Reply:
It was done below, but I changed it because it relayed on the default initialization of the `ReaderAttributes`. Also moved upwards so that it doesn't get lost in the code.